### PR TITLE
feat(atomic, headless): leverage totalNumberOfChildResults in folded results

### PR DESCRIPTION
**Pull request by alexprudhomme**
https://coveord.atlassian.net/browse/KIT-3546

### CODE REVIEW

File: packages/atomic/src/components/search/result-lists/atomic-result-children/atomic-result-children.tsx
Comment (alexprudhomme) [reply packages/atomic/src/components/search/result-lists/atomic-result-children/atomic-result-children.tsx:174]: You can only load the entire collection once. After that, we either show the entire collection using `load all` or show the initial children using `collapse`. What you are suggesting would only be useful if the source were to change as the user was loading the entire collection. 

loadedFullCollection means : _I have sent the request and got back the entire collection._

**Review by fbeaudoincoveo — APPROVED** [review]

File: packages/atomic/src/components/search/result-lists/atomic-folded-result-list/e2e/atomic-folded-result-list.e2e.ts
Comment (fpbrault) [inline packages/atomic/src/components/search/result-lists/atomic-folded-result-list/e2e/atomic-folded-result-list.e2e.ts:3]: No need to shout at us 😄 
```suggestion
test.describe('when more results are not available & there are no result children', () => {
```

File: packages/atomic/src/components/search/result-lists/atomic-folded-result-list/e2e/atomic-folded-result-list.e2e.ts
Comment (fpbrault) [inline packages/atomic/src/components/search/result-lists/atomic-folded-result-list/e2e/atomic-folded-result-list.e2e.ts:12]:
```suggestion
  test('should not show the "load all results" button', async ({
```

File: packages/atomic/src/components/search/result-lists/atomic-folded-result-list/e2e/atomic-folded-result-list.e2e.ts
Comment (fpbrault) [inline packages/atomic/src/components/search/result-lists/atomic-folded-result-list/e2e/atomic-folded-result-list.e2e.ts:20]:
```suggestion
  test('should not show result children', async ({foldedResultList}) => {
```

File: packages/atomic/src/components/search/result-lists/atomic-folded-result-list/e2e/atomic-folded-result-list.e2e.ts
Comment (fpbrault) [inline packages/atomic/src/components/search/result-lists/atomic-folded-result-list/e2e/atomic-folded-result-list.e2e.ts:25]:
```suggestion
test.describe('when more results are not available & there are result children', () => {
```

File: packages/atomic/src/components/search/result-lists/atomic-folded-result-list/e2e/atomic-folded-result-list.e2e.ts
Comment (fpbrault) [inline packages/atomic/src/components/search/result-lists/atomic-folded-result-list/e2e/atomic-folded-result-list.e2e.ts:46]:
```suggestion
  test('should not show a "no results" label', async ({foldedResultList}) => {
    await expect(foldedResultList.noResultsLabel.first()).not.toBeVisible();
  });

  test('should not show "load all results" button', async ({
    foldedResultList,
  }) => {
    await expect(
      foldedResultList.loadAllResultsButton.first()
    ).not.toBeVisible();
  });
});
```

File: packages/atomic/src/components/search/result-lists/atomic-folded-result-list/e2e/atomic-folded-result-list.e2e.ts
Comment (fpbrault) [inline packages/atomic/src/components/search/result-lists/atomic-folded-result-list/e2e/atomic-folded-result-list.e2e.ts:70]:
```suggestion
  test('should not show the "no results" label', async ({foldedResultList}) => {
```

File: packages/atomic/src/components/search/result-lists/atomic-folded-result-list/e2e/atomic-folded-result-list.e2e.ts
Comment (fpbrault) [inline packages/atomic/src/components/search/result-lists/atomic-folded-result-list/e2e/atomic-folded-result-list.e2e.ts:93]:
```suggestion
  test('should not show result children', async ({foldedResultList}) => {
    await expect(foldedResultList.resultChildren.first()).not.toBeVisible();
  });

  test('should not show the "no results" label', async ({foldedResultList}) => {
    await expect(foldedResultList.noResultsLabel.first()).not.toBeVisible();
  });
```

**Review by fpbrault — APPROVED** [review]

File: packages/atomic/src/components/search/result-lists/atomic-folded-result-list/e2e/atomic-folded-result-list.e2e.ts
Comment (alexprudhomme) [reply packages/atomic/src/components/search/result-lists/atomic-folded-result-list/e2e/atomic-folded-result-list.e2e.ts:3]: This is intended, imo without it the test names are too similar and it is confusing.

File: packages/quantic/cypress/fixtures/folded-result-list/exampleResult.json
Comment (alexprudhomme) [inline packages/quantic/cypress/fixtures/folded-result-list/exampleResult.json:53]: Instead of always being true. moreResultsAvailable now depends on the `totalNumberOfChildResults` 

image

**Review by mmitiche — APPROVED** [review]
Changes look good to me!
I will test this on my side for Quantic to double check that the folded results are displayed as expected in Quantic.

**Comment by alexprudhomme** [comment]
> Changes look good to me! I will test this on my side for Quantic to double check that the folded results are displayed as expected in Quantic.

All right, thanks, ill wait for your confirmation before merging @MehdiMitiche

**Comment by mmitiche** [comment]
@alexprudhomme Confirmed that everything related to folding is working correctly in Quantic! thank you✅

**Review by SimonMilord — APPROVED** [review]

**Pull request merged**
